src/Reflection/BetterReflection/SourceLocator/FileReadTrapStreamWrapper.php

```diff
@@ -86,7 +86,51 @@ public function stream_open($path, $mode, $options, &$openedPath): bool
 	{
 		self::$autoloadLocatedFile = $path;
 
-		return false;
+		return true;
+	}
+
+	/**
+	 * Since we allow our wrapper's stream_open() to succeed, we need to
+	 * simulate a successful read so autoloaders with require() don't explode.
+	 *
+	 * @param int $count
+	 *
+	 * @return string
+	 */
+	public function stream_read($count): string
+	{
+		// Dummy return value that is also valid PHP for require(). We'll read
+		// and process the file elsewhere, so it's OK to provide dummy data for
+		// this read.
+		return '';
```

Comment:
This causes problems when OPcache is enabled. OPcache will remember this empty string when the file is next autoloaded (which might not be within FileReadTrapStreamWrapper).

Reply:
It also causes problems when `require_once` or `include_once` is used, regardless of whether opcache is enabled or not.